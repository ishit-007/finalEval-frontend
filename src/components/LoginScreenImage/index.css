.loginImage {
  position: relative;
}
.loginImage img {
  height: 800px;
  width: 800px;
}
.loginImage .content {
  position: absolute;
  color: #4f4f4f;
  transform: translate(250px,-790px);
}
.loginImage .content h1{
  padding-top: 30px;
}
