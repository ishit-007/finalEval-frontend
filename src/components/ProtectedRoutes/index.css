.unAuthorized {
  font-family: 'Raleway', sans-serif;
  background-color: #4c3763a4;
  height: 100vh;
  padding: 10px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.unAuthorized a {
  color: white;
  text-decoration: none;
}

.unAuthorized .text {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-content: center;
  justify-content: center;
}

.unAuthorized .title {
  font-size: 5em;
  font-weight: 700;
  color: white;
}

.unAuthorized .title-2 {
  font-size: 40px;
  font-weight: 700;
  color: #0c437dc3;
}

.unAuthorized .buttons {
  margin-left: 50vw;
  width: fit-content;
  margin: 30px;
  font-weight: 700;
  border: 2px solid white;
  text-decoration: none;
  padding: 15px;
  text-transform: uppercase;
  color: white;
  border-radius: 26px;
  display: inline-block;
}
