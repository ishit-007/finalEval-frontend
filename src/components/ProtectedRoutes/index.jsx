/* eslint-disable react/no-unescaped-entities */
/* eslint-disable no-unused-vars */
import './index.css';
import axios from 'axios';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import React from 'react';

const ProtectedRoutes = ({ children }) => {
  const navigate = useNavigate();
  const [token, setToken] = React.useState(null);
  const [user, setUser] = React.useState('');
  React.useEffect(() => {
    const value = localStorage.getItem('token');
    axios
      .post('http://localhost:4000/token/validate', {
        token: value,
      })
      .then(response => {
        if (response.data) {
          setUser(response.data.userName);
          setToken(true);
        } else {
          setToken(false);
        }
      });
  }, []);

  return (
    <div>
      {token ? (
        <div>{children}</div>
      ) : (
        <div className="unAuthorized">
          <div className="text">
            <div className="title" data-content="404">
              You are not authorized to View this Page
            </div>
            <div className="title-2">Oops, You don't have permission to access this page.</div>
            <div
              className="buttons"
              onClick={() => {
                navigate('/login');
              }}>
              <a className="button" href="">
                Go to Login Page
              </a>
            </div>
          </div>
        </div>
      )}
    </div>
  );
};

ProtectedRoutes.propTypes = {
  children: PropTypes.node.isRequired,
};
export default ProtectedRoutes;
