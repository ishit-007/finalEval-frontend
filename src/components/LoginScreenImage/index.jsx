import LoginScreenImage from '../LoginScreenImage';
import { render } from '@testing-library/react';
import React from 'react';

describe('LoginScreenImage', () => {
  it('renders the component', () => {
    const { getByAltText, getByText } = render(<LoginScreenImage />);
    const image = getByAltText('Login Screen');
    const title1 = getByText(/Design API's fast,/i);
    const title2 = getByText(/Manage content easily./i);
    expect(image).toBeInTheDocument();
    expect(title1).toBeInTheDocument();
    expect(title2).toBeInTheDocument();
  });
});
