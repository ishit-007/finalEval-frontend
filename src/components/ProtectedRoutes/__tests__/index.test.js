import LoginScreenImage from '..';
import renderer from 'react-test-renderer';
import React from 'react';
import ProtectedRoutes from '..';

jest.mock('react-router-dom');

describe('ProtectedRoutes for Login', () => {
  it('should render the ProtectedRoutes component', async () => {
    const tree = renderer.create(<ProtectedRoutes text="" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
