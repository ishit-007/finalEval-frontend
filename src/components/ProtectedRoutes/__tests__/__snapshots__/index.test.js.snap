// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoutes for Login should render the ProtectedRoutes component 1`] = `
<div>
  <div
    className="unAuthorized"
  >
    <div
      className="text"
    >
      <div
        className="title"
        data-content="404"
      >
        You are not authorized to View this Page
      </div>
      <div
        className="title-2"
      >
        Oops, You don't have permission to access this page.
      </div>
      <div
        className="buttons"
        onClick={[Function]}
      >
        <a
          className="button"
          href=""
        >
          Go to Login Page
        </a>
      </div>
    </div>
  </div>
</div>
`;
